src/traces/bar/plot.js

```diff
@@ -69,12 +70,18 @@ module.exports = function plot(gd, plotinfo, cdbar) {
                         y1 = ya.c2p(p1, true);
                         x0 = xa.c2p(s0, true);
                         x1 = xa.c2p(s1, true);
+
+                        // for selections
+                        di.ct = [x1, (y0 + y1) / 2];
                     }
                     else {
                         x0 = xa.c2p(p0, true);
                         x1 = xa.c2p(p1, true);
                         y0 = ya.c2p(s0, true);
                         y1 = ya.c2p(s1, true);
+
+                        // for selections
+                        di.ct = [(x0 + x1) / 2, y1];
```

Comment:
:sos: debated logic alert :sos: 

I chose to make a bar _selected_ when the lasso/select polygon contains this point here which corresponds to the middle of the position coords at full span on the size axis.

Looking for :+1: s from @alexcjohnson @rreusser  @monfera @chriddyp @cpsievert

Reply:
That seems reasonable (especially with a 1-to-1 mapping from the (input) data value to bar), but another important question is what data do you attach to the select event when a single bar represents an aggregation of multiple values? It seems important to provide options as to how you'd like to respond to the event, so I if it were me, I would emit the raw data, the aggregated data, and the selection region (on the data scale).

Reply:
> so I if it were me, I would emit the raw data, the aggregated data, and the selection region (on the data scale).

Right now, we emit raw data and selection region. Adding the aggregated data is an interesting idea.

Reply:
And, just to be clear, assuming the model doesn't allow for partially selecting a bar that represents multiple values, I would actually vote for the selection region to cover the *entire* bar in order for a selection to be made...

The problem with that though is that it will be *very* confusing for most users if we implement two different selection criteria based on whether an aggregation is taking place

Reply:
> assuming the model doesn't allow for partially selecting a bar that represents multiple values,

Here, are you referring to histograms or bar traces with `barmode: 'stack'`?

Reply:
Furthermore, partial selection is an advanced concept that advanced users could still hack together by stacking together individual observations so that they "look aggregated"

Reply:
> Here, are you referring to histograms or bar traces with barmode: 'stack'?

Histograms

Reply:
Ok, so maybe `histogram` and `bar` should have different select logic? 

- Selecting a bar would mean encircling the most data-like pt on the bar
- Selecting an histogram item would mean encircling the entire rectangle

Reply:
> Ok, so maybe histogram and bar should have different select logic?

I'm leaning towards yes the more I think about it. I was apprehensive at first to suggest different criteria since it's potentially confusing, but the simple fact that selecting the whole bar will *always* work makes up for it

Reply:
I would offer a few arguments in favor of just keeping the "select the end" behavior:
- Some users do their own aggregation, making a `bar` trace that conceptually represents a histogram. What would we do in that case, introduce an option for whether to select the end or the whole thing?
- It would be meaningless (or at least ambiguous) to disaggregate by selecting a portion of the bar. Which sample is on top vs on bottom?
- All our other selections work by collapsing the entity to a single point to be selected
- It makes it that much harder to select just the bars you want: you could still select all the *small* bars, but you wouldn't have any way to do the likely more common act of selecting all the *largest* bars.

Reply:
I would lean towards a consistent logic for bars & hists (and maybe the possibility of adding modes in the future). I think having a different behaviour across those two charts would cause more confusion than benefits across users, especially given how often I see users creating their own "histograms" with the `bar` trace as @alexcjohnson mentioned above.

Reply:
More attributes are probably not a good idea, but perhaps the bar default would be data endpoints and the histogram default would be centroid. And if you dig deep enough to notice this subtlety, it'll lead to an attribute that can be toggled. If additional attributes had no cost, that'd be my preference, but additional attributes *do* have some small but nonzero cost (bundle size, documentation size, discoverability, maintenance, sprawl, configurability over good defaults, etc.)

Reply:
> I think having a different behaviour across those two charts would cause more confusion than benefits across users

I agree. I think there should be one simple and predictable way for this to work.

If the app developer really wants to allow users to select "partial bars" because it represents important aggregated data, then I'd argue that they should be more explicit about this in their UI and draw a rug plot below their histogram and allow users to select the exact partial sets of points that way. Or even a box plot with points or eventually a violin plot with points.

Reply:
Ok. Thanks everyone for your input. I believe the _keep-as-is_ side wins.